pkg/controller/handlers/accesscontrolrule/accesscontrolrule.go

```diff
@@ -0,0 +1,52 @@
+package accesscontrolrule
+
+import (
+	"fmt"
+
+	"github.com/obot-platform/nah/pkg/router"
+	"github.com/obot-platform/obot/apiclient/types"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+type Handler struct {
+	acrHelper *Helper
+}
+
+func New(acrHelper *Helper) *Handler {
+	return &Handler{
+		acrHelper: acrHelper,
+	}
+}
+
+func (h *Handler) PruneDeletedResources(req router.Request, _ router.Response) error {
+	acr := req.Object.(*v1.AccessControlRule)
+
+	// Make sure each resource still exists, and remove it if it is gone.
+	newResources := make([]types.Resource, 0, len(acr.Spec.Manifest.Resources))
+	for _, resource := range acr.Spec.Manifest.Resources {
+		switch resource.Type {
+		case types.ResourceTypeMCPServerCatalogEntry:
+			mcpservercatalogentry := &v1.MCPServerCatalogEntry{}
```

Comment:
nit: since we don't actually need this, it can be declared outside of the loop so we only have to allocate memory once. Same for the `mcpserver` below.

Reply:
Good call, fixed